src/source.js

```diff
@@ -140,14 +140,14 @@ class BlockedSource {
     const blockRequests = [];
 
     for (let current = firstBlockOffset; current < top; current += this.blockSize) {
-      const blockId = Math.floor(current / this.blockSize);
-      if (!this.blocks.has(blockId) && !this.blockRequests.has(blockId)) {
-        missingBlockIds.push(blockId);
+      const id = Math.floor(current / this.blockSize);
+      if (!this.blocks.has(id) && !this.blockRequests.has(id)) {
+        missingBlockIds.push({ id, signal });
```

Comment:
In order to get around the fact that fetching is highly asynchronous, where a request for one tile could send out the request for a block belonging to another tile's request, we need to store the signal for the given `fetch` call alongside the id that that `fetch` call creates.  This way, even if one tile's `fetch` call makes the request for another's `block`, the `signal` associated with remains consistent with which tile made the `fetch` call.

Reply:
This was a mistake - I did not realize a `group` could contain blocks from multiple `fetch` calls.  The code has been updated to reflect this so that any given `requestData` call depends on all possible `signals` for a `group`.